saige_assoc.py

```diff
@@ -0,0 +1,383 @@
+#     saige_command_step2 += f' --SAIGEOutputFile={saige_output_file}'
+#     saige_command_step2 += f' --chrom={chrom}'
+#     saige_command_step2 += f' --minMAF={min_maf}'
+#     saige_command_step2 += f' --minMAC={min_mac}'
+#     saige_command_step2 += f' --LOCO={loco_bool}'
+#     saige_command_step2 += f' --GMMATmodelFile={gmmat_model_path}'
+#     saige_command_step2 += f' --SPAcutoff={spa_cutoff}'
+#     saige_command_step2 += f' --varianceRatioFile={variance_ratio_path}'
+#     saige_command_step2 += f' --groupFile={group_file}'
+#     saige_command_step2 += f' --annotation_in_groupTest={group_annotation}'
+#     saige_command_step2 += f' --maxMAF_in_groupTest={max_maf_group}'
+#     saige_command_step2 += f' --minMAF_in_groupTest={min_maf_group}'
+#     # saige_command_step2 += f' --is_noadjCov={is_no_adj_cov}'
+#     # saige_command_step2 += f' --is_sparseGRM={is_sparse_grm}' # related individuals
+#     saige_command_step2 += f' --markers_per_chunk={n_markers}'
+#     saige_command_step2 += f' --pval_cutoff_for_fastTest={pval_cutoff}'
+#     # saige_command_step2 += f' --is_EmpSPA={is_emp_spa}'
+
+#     return saige_command_step2
+
+
+config = get_config()
+
+
+@click.command()
+@click.option(
+    '--pheno-cov-filename',
+    default='input/seed_1_100_nindep_100_ncell_100_lambda_2_tauIntraSample_0.5_Poisson.txt',
+)
+@click.option('--vcf-file-path', default='input/genotype_10markers.vcf.gz')
+@click.option('--covs-list', default='X1,X2,pf1,pf2')
+@click.option('--sample-covs-list', default='X1,X2')
+@click.option('--sample-id', default='IND_ID')
+@click.option(
+    '--null-output-path',
+    default='output/nindep_100_ncell_100_lambda_2_tauIntraSample_0.5_gene_1',
+)
+@click.option(
+    '--sv-output-path',
+    default='output/nindep_100_ncell_100_lambda_2_tauIntraSample_0.5_gene_1_cis',
+)
+@click.option(
+    '--gene-pvals-output-path',
+    default='output/nindep_100_ncell_100_lambda_2_tauIntraSample_0.5_gene_1_cis_genePval',
+)
+@click.option('--plink-path', default='input/n.indep_100_n.cell_1_01.step1')
+@click.option('--gene-name', default='gene_1')
+@click.option('--chrom', default='2')
+@click.option('--cis-window-file', default='input/gene_1_cis_region.txt')
+def association_pipeline(
+    pheno_cov_filename_tsv: str,
+    vcf_file_path: str,
+    covs_list: str,
+    sample_covs_list: str,
+    sample_id: str,
+    null_output_path: str,
+    sv_output_path: str,
+    gene_pvals_output_path: str,
+    plink_path: str,
+    gene_name: str,
+    chrom: str,
+    cis_window_file: str,
+    # chromosomes: str,
+    # celltypes: str,
+):
+    """
+    Run association for one gene
+    """
+
+    sb = hb.ServiceBackend(
+        billing_project=get_config()['hail']['billing_project'],
+        remote_tmpdir=remote_tmpdir(),
+    )
+    batch = hb.Batch('SAIGE-QTL pipeline', backend=sb)
+
+    # step 1 (fit null)
+    fit_null_job = batch.new_job(name='fit null')
+    fit_null_job.image(SAIGE_QTL_IMAGE)
```

Comment:
these jobs don't define any attached storage, do they write output files?

Reply:
what do you mean by this? And yes they do write output files

Reply:
you aren't adding any hard drive space to these jobs (e.g. `job.storage('10Gi')`), so the hail batch job may fail if the scripts try to write any files. If they only write to GCP you might be ok, but if they write anything locally you will need some local storage capacity

Reply:
right sorry i dont remember having to do this before! i think with test data everything should be tiny but i guess it doesnt hurt to ask for some storage! I think it goes back a bit to another answer I have and whether having this pipeline run first with test data and then with real data should actually be two separate scripts?